libraries/default.rb

```diff
@@ -127,18 +127,19 @@ def private_recipe_ips(cookbook, recipe)
     def resolve_hostname(ip)
       require 'resolv'
       hostf = Resolv::Hosts.new
-      dns = Resolv::DNS.new
-      # Try and resolve hostname first using /etc/hosts, then use DNS
+
       begin
-        hostname = hostf.getname(ip)
+        hname = hostf.getname(ip)
       rescue
         begin
-          hostname = dns.getname(ip)
+          hname = node['fqdn']
         rescue
-          raise "Cannot resolve the hostname for IP address: #{ip}"
+            raise "Cannot resolve the hostname for IP address: #{h}"
```

Comment:
When the exception is raised, `h` does not exist

Reply:
thanks